src/app.ts

```diff
@@ -13,14 +13,49 @@ interface TemperatureSummary {
   average: number
 }
 
+
+const redingMap: { [key: string]: { [key: string]: number[] } } = {}; 
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  readings.forEach((reding) => {
+
+    const date = reding.time.toString().split('T')[0];
```

Comment:
Is it worth it declare an external function for this date string splitting since you are using it twice?

Reply:
Also is the .split() function needed? it would works if you remove it?

Reply:
1. I did it with an external function pointing so I can use the code later.  Because if I needed to modify it in the future, I would only have to do it in one part and not search for each function one by one.

2.In this case it was not strictly necessary to use the split because the date was not in ISO format, but I had previously worked with dates in that format and that is why I used it. It should continue working, only it returns the full date as received from the backend or in this case it is a static or "burned" date.

Reply:
1.- I mean you can define an external function like this, I'm asking you if it's worth it to do it or not

```
const formatDate = (date : Date) => {
     return date.toString().split('T')[0];
 }
```

Reply:
2.- So you were worried about time formatting, is there a better way to do that? maybe I`ntl.DateTimeFormat`?

Reply:
1. In this case, since the function is only used once and the format is not expected to change, I don't think it's worth declaring an external function... 

2. I had not worked with intl.DateTimeFormat before, that's why I didn't consider it. Thanks for the comment, I'm going to read more about it for date formatting!